From 479b3c8ee1816ac8019c202398de9fc93bf109ca Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Daniel=20Ekl=C3=B6f?= <daniel@ekloef.se>
Date: Wed, 5 Apr 2023 14:39:02 +0200
Subject: [PATCH] *.desktop: add StartupWMClass=foot

At least Gnome needs this in order to link running instances of foot
to their corresponding .desktop file, used e.g. when determining which
icon to display for running applications.

Closes #1317
---
 CHANGELOG.md                          | 6 ++++++
 org.codeberg.dnkl.foot-server.desktop | 1 +
 org.codeberg.dnkl.foot.desktop        | 1 +
 org.codeberg.dnkl.footclient.desktop  | 1 +
 4 files changed, 9 insertions(+)

diff --git a/org.codeberg.dnkl.foot-server.desktop b/org.codeberg.dnkl.foot-server.desktop
index 6e8891c0..a40117c7 100644
--- a/org.codeberg.dnkl.foot-server.desktop
+++ b/org.codeberg.dnkl.foot-server.desktop
@@ -9,3 +9,4 @@ Keywords=shell;prompt;command;commandline;
 Name=Foot Server
 GenericName=Terminal
 Comment=A wayland native terminal emulator (server)
+StartupWMClass=foot
diff --git a/org.codeberg.dnkl.foot.desktop b/org.codeberg.dnkl.foot.desktop
index f072568d..720d35a9 100644
--- a/org.codeberg.dnkl.foot.desktop
+++ b/org.codeberg.dnkl.foot.desktop
@@ -9,3 +9,4 @@ Keywords=shell;prompt;command;commandline;
 Name=Foot
 GenericName=Terminal
 Comment=A wayland native terminal emulator
+StartupWMClass=foot
diff --git a/org.codeberg.dnkl.footclient.desktop b/org.codeberg.dnkl.footclient.desktop
index f82f282b..dc8bc5dc 100644
--- a/org.codeberg.dnkl.footclient.desktop
+++ b/org.codeberg.dnkl.footclient.desktop
@@ -9,3 +9,4 @@ Keywords=shell;prompt;command;commandline;
 Name=Foot Client
 GenericName=Terminal
 Comment=A wayland native terminal emulator (client)
+StartupWMClass=foot
